template/src/ontology/run.sh.jinja2

```diff
@@ -23,10 +30,20 @@ if [ x$ODK_DEBUG = xyes ]; then
     TIMECMD="/usr/bin/time -f ### DEBUG STATS ###\nElapsed time: %E\nPeak memory: %M kb"
 fi
 
-docker run -v $PWD/../../:/work -w /work/src/ontology -e ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS" -e JAVA_OPTS="$ODK_JAVA_OPTS" {% if project.use_external_date is sameas True %}-e TODAY=`date +%Y-%m-%d` {% endif %}--rm -ti obolibrary/$IMAGE $TIMECMD "$@"
+VOLUME_BIND=$PWD/../../:/work
+WORK_DIR=/work/src/ontology
+
+if [ -n "$USE_SINGULARITY" ]; then
+    export ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS"
```

Comment:
I am not very familiar with singularity.. So contrary to `docker`, you set the environment variables in the main environment, and somehow the singularity container has access to these globally set variables? In ODK its necessary to set these parameters inside the running container..

Reply:
Well, on all the singularity setups I have used in 4 different HPCs, they normally inherit all variables from the parent environment (which is sometimes problematic as well).

Reply:
Thanks for the clarification!

Reply:
(Just out of curisioty, _can_ you set th variables inside the container?)

Reply:
According to the documentation, yes. It should be possible to prevent Singularity from inheriting variables from the underlying system with `--cleanenv` and instead to explicitly pass _only_ the variables we need with `--env VAR1=value1,VAR2=value2`.

(I can’t test that today as I am on the move and my laptop is not properly configured to run Singularity.)

Reply:
Actually, yes, this works:

```sh
singularity exec --cleanenv --env "ROBOT_JAVA_ARGS=$ODK_JAVA_OPTS,JAVA_OPTS=$ODK_JAVA_OPTS" --bind $VOLUME_BIND -W $WORK_DIR docker://obolibrary/$IMAGE:$TAG $TIMECMD "$@"
```

Reply:
Why do we still need the `export` commands then?

Reply:
Yes, not needed, have removed them now.